### ISSUE TITLE
"empty_statements" doesn't know about case fallthrough

### ISSUE THREAD

**Issue by Hixie** [dart-lang/linter#4410]
I had this code:

```dart
      switch (customization.touchFeedback) {
        case HapticStyle.none:
          ;
        case HapticStyle.vibrate:
          HapticFeedback.vibrate();
        // ...
```

...and the `;` line got the message "Unnecessary empty statement", but if I remove it, it changes the semantics of the code.

**Comment by lrhn** [comment]
Sometimes you do want an empty statement. I think the link description should recognize that and suggest using `{}` for those cases. 
Then I'd still warn about the `;` here, but it's not because the empty statement is unnecessary, but because it's not `{}`. 
(This is still a bug, the *message* is incorrect.)

Also `case Foo: {}` doesn't read particularly well, I understand why you'd use `;` instead, but here I'd actually recommend `break;` as the no-op case body. It's unmistakable.

**Comment by bwilkerson** [comment]
I also prefer `break;` because I think it's more specific, but I'm not convinced that the lint should be flagging an empty statement in this location. The purpose of the lint is to find places where an empty statement is either likely to be causing a bug or is unnecessary. I'm not sure this qualifies.

But if we do decide to leave the lint flagging such locations, we should consider ensuring that we don't offer a fix to remove the statement, but do offer a fix to replace it with a `break` statement. I'd also be ok offering both fixes.

And yes, I like the idea of offering a fix to convert `;` into `{}` in other locations.

**Comment by Hixie** [comment]
I also prefer `break`. My concern is just that the message can lead to accidentally causing a significant semantic change to the code that is really hard to spot in retrospect. If you blindly remove all the "excessive" semicolons, then try to figure out why your code broke, there's no obvious remnant in the code to suggest the magnitude of the change that happened.

**Comment by pq** [comment]
My sense is that this is currently a false positive and should just not be flagging this case.

As for preferring `break` that feels like another lint. If anyone feels like jotting down a [proposal](https://github.com/dart-lang/linter/issues/new?assignees=&labels=lint+proposal%2C+status%3A+pending&projects=&template=lint-proposal.md&title=proposal%3A+%60%3Crule_name%3E%60), I bet we could get pretty quick consensus and get it implemented.
